build-it/build.py

```diff
@@ -69,22 +83,40 @@ def copy_binaries_for_target(self):
             os.makedirs(generator_dll_destination)
         distutils.file_util.copy_file(generator_dll_source, generator_dll_destination)
 
-    def copy_built_binaries(self):
-        if self.build_target == "All":
-            self.copy_binaries_all_targets()
+    def copy_built_binaries(self, args):
+        if args.target == "All":
+            self.copy_binaries_all_targets(args)
         else:
             self.copy_binaries_for_target()
 
 
-    def build(self):
-        if not self.build_target == "All" and not self.vipb_build_only:
-            self.cpp_build()
-        self.copy_built_binaries()
+    def build(self, args):
+        if not args.target == "All" and not args.buildcpp:
+            self.cpp_build(args)
+        self.copy_built_binaries(args)
         build_vi_path = os.path.join(self.build_script_directory, "LV Build", "BuildGRPCPackages.vi")
-        subprocess.run(["LabVIEWCLI", "-OperationName", "RunVI", "-VIPath", build_vi_path, os.path.join(self.root_directory, "labview source")])
+        build_vipkgs = subprocess.run(["LabVIEWCLI", "-OperationName", "RunVI", "-VIPath", build_vi_path, os.path.join(self.root_directory, "labview source")], capture_output = True)
+        if (build_vipkgs.returncode != 0):
+            raise Exception(f'Failed to Build vipkgs { build_vipkgs.stderr.decode() }')
 
 def main():
     gRPCPackageBuilder = LVgRPCBuilder()
-    gRPCPackageBuilder.build()
+    args = gRPCPackageBuilder.parse_args()
+    
+    if args.library_version != "" :
+        vipb_files = vipb_helper_class.get_vipb_files(gRPCPackageBuilder.root_directory)
+        for vipb_file in vipb_files:
+            vipb_helper_class.update_vipb_verion(vipb_file = vipb_file, library_version = args.library_version)
+        # build_vipb will be -1 when we donot want to build the vipb files    
+        if args.build_vipb == -1:
+            return
```

Comment:
It may be better to split the update workflow into a separate `update_vipb_version.py`. Having a mode for a `build.py` that doesn't do any build seems odd.

Reply:
Done